src/test/unit/services/pathfinder/normal_glm_test.cpp

```diff
@@ -0,0 +1,297 @@
+    tmp.row(1) = std::get<1>(diagnostics.optim_path_[i]);
+    // std::cout << "Iter: " << i << "\n" << tmp << "\n";
+  }
+  Eigen::IOFormat CommaInitFmt(Eigen::StreamPrecision, 0, ", ", ", ", "\n", "",
+                               "", "");
+
+  // std::cout << "---- Results  -------" << std::endl;
+  Eigen::VectorXd mean_vals = param_vals.rowwise().mean().eval();
+  //       std::cout << "Mean Values: \n" <<
+  //       mean_vals.transpose().eval().format(CommaInitFmt) << "\n";
+  Eigen::VectorXd sd_vals = (((param_vals.colwise() - mean_vals)
+                                  .array()
+                                  .square()
+                                  .matrix()
+                                  .rowwise()
+                                  .sum()
+                                  .array()
+                              / (param_vals.cols() - 1))
+                                 .sqrt())
+                                .transpose()
+                                .eval();
+  // std::cout << "\n\n" << param_vals.format(CommaInitFmt) << "\n\n";
+  // std::cout << "rows: " << param_vals.rows() << " cols: " <<
+  // param_vals.cols() << "\n";
+  Eigen::MatrixXd prev_param_vals = stan::test::normal_glm_param_vals();
+  // std::cout << "\n\n" << prev_param_vals.format(CommaInitFmt) << "\n\n";
+  Eigen::VectorXd prev_mean_vals = prev_param_vals.rowwise().mean().eval();
+  //       std::cout << "Mean Values: \n" <<
+  //       mean_vals.transpose().eval().format(CommaInitFmt) << "\n";
+  Eigen::VectorXd prev_sd_vals = (((prev_param_vals.colwise() - prev_mean_vals)
+                                       .array()
+                                       .square()
+                                       .matrix()
+                                       .rowwise()
+                                       .sum()
+                                       .array()
+                                   / (prev_param_vals.cols() - 1))
+                                      .sqrt())
+                                     .transpose()
+                                     .eval();
+  Eigen::MatrixXd ans_diff = param_vals - prev_param_vals;
+  Eigen::VectorXd mean_diff_vals = ans_diff.rowwise().mean();
+  //      std::cout << "diff Mean Values: \n" <<
+  //      mean_diff_vals.transpose().eval().format(CommaInitFmt) << "\n";
+  Eigen::VectorXd sd_diff_vals = (((ans_diff.colwise() - mean_diff_vals)
+                                       .array()
+                                       .square()
+                                       .matrix()
+                                       .rowwise()
+                                       .sum()
+                                       .array()
+                                   / (ans_diff.cols() - 1))
+                                      .sqrt())
+                                     .transpose()
+                                     .eval();
+
+  Eigen::MatrixXd all_mean_vals(3, 10);
```

Comment:
why put this into a 3 x 10 array rather than using 3 separate arrays?  We found it a bit hard to read this way.

Reply:
Mostly just because it's easier to print to the console so I could check that the values all lined up